plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java

```diff
@@ -0,0 +1,421 @@
+                    "   primary_key = ARRAY['orderkey']\n" +
+                    ")";
+            String tableName = testTable.getName();
+            assertThat((String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue())
+                    .isEqualTo(format(pattern, catalog, schema, tableName));
+        }
+    }
+
+    @Test
+    public void testAvgDecimalExceedingSupportedPrecision()
+    {
+        try (TestTable testTable = new TestTable(
+                getQueryRunner()::execute,
+                "test_avg_decimal_exceeding_supported_precision",
+                "(a decimal(38, 38), b bigint)",
+                List.of(
+                        "CAST ('0.12345671234567123456712345671234567121' AS decimal(38, 38)), 1",
+                        "CAST ('0.12345671234567123456712345671234567122' AS decimal(38, 38)), 2",
+                        "CAST ('0.12345671234567123456712345671234567123' AS decimal(38, 38)), 3",
+                        "CAST ('0.12345671234567123456712345671234567124' AS decimal(38, 38)), 4",
+                        "CAST ('0.12345671234567123456712345671234567125' AS decimal(38, 38)), 5",
+                        "CAST ('0.12345671234567123456712345671234567126' AS decimal(38, 38)), 6",
+                        "CAST ('0.12345671234567123456712345671234567127' AS decimal(38, 38)), 7"))) {
+            assertThat(query("SELECT avg(a) avg_a  FROM " + testTable.getName()))
+                    .matches("SELECT CAST ('0.12345671234567123456712345671234567124' AS decimal(38, 38))");
+            assertThat(query(format("SELECT avg(a) avg_a FROM %s WHERE b <= 2", testTable.getName())))
+                    .matches("SELECT CAST ('0.123456712345671234567123456712345671215' AS decimal(38, 38))");
+        }
+    }
+
+    @Override
+    protected TestTable createAggregationTestTable(String name, List<String> rows)
+    {
+        return new TestTable(
+                onRemoteDatabase(),
+                name,
+                "(short_decimal decimal(9, 3), long_decimal decimal(30, 10), t_double double, a_bigint bigint primary key)",
+                rows);
+    }
+
+    @Override
+    protected TestTable createTableWithDoubleAndRealColumns(String name, List<String> rows)
+    {
+        return new TestTable(
+                onRemoteDatabase(),
+                name,
+                "(t_double double, u_double double, v_real real, w_real real primary key)",
+                rows);
+    }
+
+    @Override
+    protected Optional<String> filterColumnNameTestData(String columnName)
+    {
+        Set<String> unsupportedColumnNames = ImmutableSet.of(
+                "a.dot", // https://issues.apache.org/jira/browse/IGNITE-18102
+                "a\"quote",
+                "a`backtick`",
+                "a/slash`",
+                "a\\backslash`");
+        if (unsupportedColumnNames.contains(columnName)) {
+            return Optional.empty();
+        }
+
+        return Optional.of(columnName);
+    }
```

Comment:
Please override `isColumnNameRejected` instead of changing to empty.

Reply:
Updated the `show create table` using `orders` table, just saw phoenix do in this way.